File: Hackathon/Hackathon.Service/BoardItems/BoardItemService.cs
using System.Collections.Generic;
using System.Linq;
using Hackathon.Core.Models.Board.Enums;

namespace Hackathon.Service.BoardItems
{
    using Core.Models.Board;
    using Repository.DataAccess;

    public interface IBoardItemService
    {
        void SaveBoardItems(BoardState itemsToSave);
        void SaveBoardItem(BoardItem itemToSave);
        BoardState GetBoardItems();
    }

    public class BoardItemService : IBoardItemService
    {

        private BoardItemRepository BoardItemRepository { get; }

        public BoardItemService()
        {
            BoardItemRepository = new BoardItemRepository();
        }

        public BoardState GetBoardItems()
        {
            var items = BoardItemRepository.SelectBoardItems();
            return new BoardState
            {
                Open = items.Where(item => item.Status == BoardStatus.Open).ToList(),
                Progress = items.Where(item => item.Status == BoardStatus.Progress).ToList(),
                Blocked = items.Where(item => item.Status == BoardStatus.Blocked).ToList(),
                Done = items.Where(item => item.Status == BoardStatus.Done).ToList()
            };
        }

        public void SaveBoardItem(BoardItem itemToSave)
        {
            BoardItemRepository.CreateBoardItems(new[] { itemToSave});
        }

        public void SaveBoardItems(BoardState itemsToSave)
        {
            BoardItemRepository.UpdateBoardItems(itemsToSave.Done.Union(itemsToSave.Open).Union(itemsToSave.Blocked).Union(itemsToSave.Progress));
        }

    }
}

File: Hackathon/Hackathon.Web/client/services/boardItemService.js
(function (angular) {
    angular.
        module('board').
        service('boardItemService', boardItemService);

    boardItemService.$inject = ['$http'];

    function boardItemService($http) {
        
        var service = {
            getBoardItems: getBoardItems,
            getBoardItemsForUser: getBoardItemsForUser,
            saveBoardItem: saveBoardItem,
            saveBoardItems: saveBoardItems
        };

        function getBoardItems() {
            return  {
                selected: null,
                lists: { 'open': [], 'progress': [], 'blocked': [], 'done': [] }
            };
        }

        function getBoardItemsForUser(userId) {
            return $http.get('api/task/' + userId).then(success, error);

            function success(response) {
                return response;
            }

            function error(response) {
                console.log('Didnt work from server');
            }
        }

        function saveBoardItem(boardItem) {
            return $http.put('api/task', boardItem).then(success, error);

            function success(response) {
                return response;
            }

            function error(response) {
                console.log('Unable to save board item.');
            }
        }

        function saveBoardItems(items) {
            return $http.post('api/task', items).then(success, error);

            function success(response) {
                console.log('great success');
            }

            function error(response) {
                
            }
        }

        return service;
    }

})(window.angular);